src/item.h

```diff
@@ -175,6 +175,8 @@ inline bool is_crafting_component( const item &component );
 class item : public visitable<item>
 {
     public:
+        using FlagsSetType = std::set<std::string>;
```

Comment:
Any particular reason you're switching back to `std::set` in this PR?  When I last profiled, `cata::flat_set` was faster.  I realise you're planning to change this again soon, but it seems odd to change it here.

Reply:
That's actually the topic I wanted to discuss with you. If you have time, could you please check out #44694 and run `flags_benchmark` (from generic_factory_test.cpp) on your system? On my system profiling and benchmarks show that `std::set` is slightly faster than `cata::flat_set` for lookups in scenarios that mimic real flag usage. That seems counterintuitive and I might be missing something.

Reply:
There was a particular test I was using to benchmark when I switched it to `flat_set`; I can't remember now which test it was, but it was one of the slow ones; maybe `starting_items`?

I haven't tried running your `flags_benchmark`, but I'm going to make a wild conjecture: The benefit of `flat_set` is primarily that it uses fewer cache lines.  This benefit is only relevant when there's lots of other stuff going on.  In `flags_benchmark` it looks like you're doing nothing but messing with flags, so everything will fit in L1 cache and I'd expect the performance to be pretty much the same.  But it might still be faster to use `flat_set` in the context of running the rest of the game, when the `set` elements keep getting pushed out of cache.

Reply:
I agree with your theory. I had the cache in mind when I was writing that benchmark, because [for sections like this](https://github.com/CleverRaven/Cataclysm-DDA/blob/ca4148daec284a19377da16f0fbb2cb4b3206da9/src/item.cpp#L9989) the same set is accessed repeatedly and is probably cached. 
However, I also agree, that changing the type of collection is outside of the scope of this specific PR, so I'll change it back to `flat_set`.